### PR TITLE
Fix path output for required properties

### PR DESCRIPTION
**Pull request by JustTSK**
Example:

Before:
   `day: "The property day is required"`
After
   `availability[0].hours[0].day: "The property day is required"`

### CODE REVIEW

**Comment by bighappyface** [comment]
@JustTSK thanks for the PR.

Please remove the `.gitignore` addition in favor of your [global git ignore](http://stackoverflow.com/questions/7335420/global-git-ignore) for `.DS_Store`.

I think the outcome is good, but the implementation could be cleaner.

Perhaps something like:

```php
$required = (!$path) ?: "$path.$required";
```

This would allow you to save a variable, a full condition block, and use the more efficient string interpolation over concatenation.

@localheinz @samkeen thoughts?

File: src/JsonSchema/Constraints/UndefinedConstraint.php [hunk 1]
```diff
@@ -122,7 +122,12 @@ protected function validateCommonProperties($value, $schema = null, $path = null
                 // Draft 4 - Required is an array of strings - e.g. "required": ["foo", ...]
                 foreach ($schema->required as $required) {
                     if (!property_exists($value, $required)) {
-                        $this->addError($required, "The property " . $required . " is required");
+                        if($path) {
```
Comment (GrahamCampbell) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: cs

**Comment by JustTSK** [comment]
@bighappyface we cant override `$required` variable because its used in error description: `"The property " . $required . " is required"`

File: src/JsonSchema/Constraints/UndefinedConstraint.php [hunk 2]
```diff
@@ -122,7 +122,12 @@ protected function validateCommonProperties($value, $schema = null, $path = null
...
+                            $property_path = $path .".". $required;
+                        } else {
+                            $property_path = $required;
+                        }
+                        $this->addError($property_path, "The property " . $required . " is required");
```
Comment (JustTSK) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: can be
```php
$this->addError( (!$path) ? "$required" : "$path.$required" , "The property " . $required . " is required");
```

File: src/JsonSchema/Constraints/UndefinedConstraint.php [hunk 3]
```diff
@@ -122,7 +122,7 @@ protected function validateCommonProperties($value, $schema = null, $path = null
                 // Draft 4 - Required is an array of strings - e.g. "required": ["foo", ...]
                 foreach ($schema->required as $required) {
                     if (!property_exists($value, $required)) {
-                        $this->addError($required, "The property " . $required . " is required");
+                        $this->addError( (!$path) ? $required : "$path.$required" , "The property " . $required . " is required");
```
Comment (GrahamCampbell) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: cs

Comment (JustTSK) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: ?

Comment (bighappyface) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: Code style

**Comment by localheinz** [comment]
@JustTSK 

Build is currently failing because of some weird error:

> [Composer\Repository\RepositorySecurityException]                            
The contents of http://packagist.org/p/provider-2015-04$09177ee9742784c520bc46745e536dd726464fd578db4be65f2924bf8c6f1c7c.json do not match its signature. This should indicate a man-in-the-middle attack. Try running composer again and report this if you think it is a mistake.

File: src/JsonSchema/Constraints/UndefinedConstraint.php (hunk 3 above)
Comment (JustTSK) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: Ok, can i have more info what code style are you using? I cant find it anywhere in the docs

Comment (JustTSK) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: @GrahamCampbell ?

**Comment by mwaterboehr** [comment]
I have not noticed this pull request before. So I just created another pull request (PR #190) that fixes this issue. What do you think about my solution there?

**Comment by JustTSK** [comment]
Updated pull request

**Pull request closed without merging**